pkg/scheduler/internal/cache/cache.go

```diff
@@ -554,13 +556,17 @@ func (cache *schedulerCache) AddNode(node *v1.Node, resourceProviderId string) e
 	return n.info.SetNode(node)
 }
 
-func (cache *schedulerCache) UpdateNode(oldNode, newNode *v1.Node, resourceProviderId string) error {
+func (cache *schedulerCache) UpdateNode(oldNode, newNode *v1.Node, nodeListers map[string]corelisters.NodeLister) error {
 	cache.mu.Lock()
 	defer cache.mu.Unlock()
 
 	n, ok := cache.nodes[newNode.Name]
 	if !ok {
 		nodeInfo := schedulernodeinfo.NewNodeInfo()
+		_, resourceProviderId, err := nodeutil.GetNodeFromNodelisters(nodeListers, newNode.Name)
```

Comment:
under what situation could the old.rpid be not the the same as new.rpid? if they might be different, this lookup is critical.

Reply:
This case is when node not found in our cache. Since the cache was flat, not found by node name means the node is not in the cache. Therefore, we won't get old rpId.

I think this case happen if we missed AddNode event. Perhaps just a safety guard.